fhirAssembly/src/test/java/io/vantiq/extsrc/fhirAssembly/TestFhirUseAccessToken.java

```diff
@@ -0,0 +1,304 @@
+        Map<String, ?> config =  Map.of("fhirServerBaseUrl", fauxServer,
+                                        "authenticationMechanism", "ApplicationManaged",
+                                        "basicAccessToken", tokenValue);
+        doOperations("Bearer", "IAmNotATeaPot", config);
+    }
+    
+    @Test
+    public void testOperationsWithOverrideConfigBasedTokenAndRealm() {
+        String tokenValue = "ImALittleTeaPot";
+        String tokenRealm = "HonaleeLandOfPuff";
+        Map<String, ?> config =  Map.of("fhirServerBaseUrl", fauxServer,
+                                        "authenticationMechanism", "ApplicationManaged",
+                                        "basicRealm", tokenRealm,
+                                        "basicAccessToken", tokenValue);
+        doOperations("LandOfMakeBelieve", "IAmNotATeaPot", config);
+    }
+    
+    @Test
+    public void testOperationsWithAppManagedConfigBasedToken() {
+        String tokenValue = "ImALittleTeaPot";
+        Map<String, ?> config =  Map.of("fhirServerBaseUrl", fauxServer,
+                                        "authenticationMechanism", "ApplicationManaged",
+                                        "basicAccessToken", tokenValue);
+        doOperations("Bearer", tokenValue, config, true);
+    }
+    
+    @Test
+    public void testOperationsWithBasicTokenAndRealm() {
+        String tokenValue = "ImALittleTeaPot";
+        String tokenRealm = "HonaleeLandOfPuff";
+        Map<String, ?> config =  Map.of("fhirServerBaseUrl", fauxServer,
+                                        "authenticationMechanism", "Basic",
+                                        "basicRealm", tokenRealm,
+                                        "basicAccessToken", tokenValue);
+        doOperations(tokenRealm, tokenValue, config, true);
+    }
+    
+    @Test
+    public void testOperationsBasicConfigBasedToken() {
+        String tokenValue = "ImALittleTeaPot";
+        Map<String, ?> config =  Map.of("fhirServerBaseUrl", fauxServer,
+                                        "authenticationMechanism", "Basic",
+                                        "basicAccessToken", tokenValue);
+        doOperations("Bearer", tokenValue, config, true);
+    }
+    
+    @Test
+    public void testOperationsWithAppManagedConfigBasedTokenAndRealm() {
+        String tokenValue = "ImALittleTeaPot";
+        String tokenRealm = "HonaleeLandOfPuff";
+        Map<String, ?> config =  Map.of("fhirServerBaseUrl", fauxServer,
+                                        "authenticationMechanism", "ApplicationManaged",
+                                        "basicRealm", tokenRealm,
+                                        "basicAccessToken", tokenValue);
+        doOperations(tokenRealm, tokenValue, config, true);
+    }
+    
+    
+    void doOperations(String tType, String tValue, Map<String, ?> assyConfig) {
+        doOperations(tType, tValue, assyConfig, false);
+    }
+    
```

Comment:
Make a series of so-called FHIR calls that span the different underlying call mechanisms we use.  Our code doesn't interpret the FHIR Json coming back, so just passing back the map of the headers is fine here.  And we use those to check that we send what was expected.